sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/CentralMomentAgg.scala

```diff
@@ -62,6 +63,9 @@ abstract class CentralMomentAgg(child: Expression)
   protected val m3 = AttributeReference("m3", DoubleType, nullable = false)()
   protected val m4 = AttributeReference("m4", DoubleType, nullable = false)()
 
+  protected val divideByZeroEvalResult: Expression =
+    if (SQLConf.get.legacyStatisticalAggregate) Double.NaN else Literal.create(null, DoubleType)
```

Comment:
can we move the flag to constructor parameter? e.g.
```
abstract class CentralMomentAgg(child: Expression, nullOnDivideByZero: Boolean) {
  ...
  protected def divideByZeroEvalResult: Expression = if (nullOnDivideByZero) ...
}
```

Reply:
done